src/app/api/serializers/transfer.py

```diff
@@ -0,0 +1,32 @@
+from rest_framework import serializers
+
+from app.domain.models.transfer import Transfer
+
+
+class TransferSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Transfer
+        fields = "__all__"
+
+
+class TransferMeSerializer(serializers.ModelSerializer):
+    origin = serializers.SerializerMethodField()
+    destiny = serializers.SerializerMethodField()
```

Comment:
Você poderia utilizar o argumento `source` do serializer

> https://www.django-rest-framework.org/api-guide/fields/#source

```suggestion
    origin = serializers.CharField(source='origin.user.full_name')
    destiny = serializers.CharField(source='destiny.user.full_name')
```